src/panels/config/devices/device-detail/integration-elements/zwave_js/device-actions.ts

```diff
@@ -43,59 +43,68 @@ export const getZwaveDeviceActions = async (
 
   const nodeStatus = await fetchZwaveNodeStatus(hass, device.id);
 
-  if (!nodeStatus || nodeStatus.is_controller_node) {
+  if (!nodeStatus) {
     return [];
   }
 
-  const actions = [
-    {
-      label: hass.localize(
-        "ui.panel.config.zwave_js.device_info.device_config"
-      ),
-      icon: mdiCog,
-      href: `/config/zwave_js/node_config/${device.id}?config_entry=${entryId}`,
-    },
-    {
-      label: hass.localize(
-        "ui.panel.config.zwave_js.device_info.reinterview_device"
-      ),
-      icon: mdiChatQuestion,
-      action: () =>
-        showZWaveJSReinterviewNodeDialog(el, {
-          device_id: device.id,
-        }),
-    },
-    {
-      label: hass.localize("ui.panel.config.zwave_js.device_info.heal_node"),
-      icon: mdiHospitalBox,
-      action: () =>
-        showZWaveJSHealNodeDialog(el, {
-          device,
-        }),
-    },
-    {
-      label: hass.localize(
-        "ui.panel.config.zwave_js.device_info.remove_failed"
-      ),
-      icon: mdiDeleteForever,
-      action: () =>
-        showZWaveJSRemoveFailedNodeDialog(el, {
-          device_id: device.id,
-        }),
-    },
-    {
-      label: hass.localize(
-        "ui.panel.config.zwave_js.device_info.node_statistics"
-      ),
-      icon: mdiInformation,
-      action: () =>
-        showZWaveJSNodeStatisticsDialog(el, {
-          device,
-        }),
-    },
-  ];
+  const actions: DeviceAction[] = [];
+
+  if (!nodeStatus.is_controller_node) {
```

Comment:
Why this extra if, if we are only adding actions when not `is_controller_node`

Reply:
we are adding actions if there is a controller node just for the firmware update

Reply:
is this possibly the confusing part because I moved the not operator outside of each individual check? I can move it back inside so each individual statement is easier to understand: https://github.com/home-assistant/frontend/pull/15515/files#diff-4834eb73ab93f8dfa87c03847b42ab8539d6b51909e48038a784dc80d08e2472R103-R106